src/TLuaInterpreter.cpp

```diff
@@ -16057,3 +16058,27 @@ int TLuaInterpreter::removeCommandLineMenuEvent(lua_State * L)
     lua_pushboolean(L, true);
     return 1;
 }
+
+int TLuaInterpreter::deleteMap(lua_State* L)
+{
+    Host& host = getHostFromLua(L);
+
+    if (!host.mpMap || !host.mpMap->mpRoomDB) {
+        // These tests pass even if the map is empty, however there can still
+        // be deleteable data present even in that case - and this test will
+        // still succeed immediately after this function has been used!
+        return warnArgumentValue(L, __func__, "no map present or loaded");
+    }
+
+    host.mpMap->mapClear();
+    // We do not normally produce an on-screen message for Lua API actions but
+    // this function might confuse the user if it is done from a script/package
+    // without them knowing about it:
+    host.postMessage(tr("[  OK  ]  - Any existing map and its data has been destroyed by a Lua function call."));
```

Comment:
Yeah, either that, or this message will confuse them then. 😄 Let's put the text in debug console as usual.

Also please rephrase "Any existing map and its data has been destroyed by a Lua function call."
How about "deleteMap: script has deleted all map data."